src/opm/parser/eclipse/EclipseState/Schedule/Schedule.cpp

```diff
@@ -523,6 +525,33 @@ void Schedule::iterateScheduleSection(std::size_t load_start, std::size_t load_e
         this->snapshots.back().actions.update( std::move(new_actions) );
     }
 
+    void Schedule::inspect_actionx_keyword(const ScheduleGrid& grid, const DeckKeyword& keyword){
+        static std::unordered_set<std::string> keyword_list = {"COMPDAT"};
+
+        if(keyword_list.count(keyword.name())){
```

Comment:
The code here is set up to work identically with all keywords from the keyword_list; that is not correct. You have now written a correct implementation for the COMPDAT keyword - other keywords will be different. Something like:
```
if (keyword_list.count(keyword.name()) == 0)
    return;

if (keyword.name() == "COMPDAT") {
    // Continue with the function you have wriitten
```